src/modules/inventing/parseRecipe.js

```diff
@@ -0,0 +1,43 @@
+import getText from '../common/getText';
+import querySelector from '../common/querySelector';
+import querySelectorArray from '../common/querySelectorArray';
+
+export function parseIngredientTable(table) {
+  const [have, need] = getText(querySelector('tr:nth-child(2) td', table)).split('/').map((v) => parseInt(v, 10));
+  return Object({
+    id: querySelector('img', table).src.match(/(\d+)\.[A-Za-z]+$/)[1],
+    have,
+    need,
+  });
+}
+
+export function ingredients() {
+  const componentTables = querySelectorArray(
+    '#pCC > table:nth-child(1) >'
```

Comment:
this seems brittle. could you hook into the cell background image instead?

Reply:
Would this be better?
```
 const ingredientTables = querySelectorArray(
    '#pCC td[background^="https://cdn2.fallensword.com/ui/inventory/"]',
  ).map((td) => td.parentElement.parentElement);
```

Reply:
yup that is pretty much what i had in mind.
your version works better though because you can treat items and components as a single array, which in turn will make findMaxInv simpler.